modules/compiler/vecz/source/vectorization_context.cpp

```diff
@@ -368,6 +374,227 @@ Function *VectorizationContext::getOrCreateMaskedFunction(CallInst *CI) {
   return newFunction;
 }
 
+std::optional<VectorizationContext::MaskedAtomicRMW>
+VectorizationContext::isMaskedAtomicRMWFunction(const Function &F) const {
+  auto VFInfo = decodeVectorizedFunctionName(F.getName());
+  if (!VFInfo) {
+    return std::nullopt;
+  }
+  auto [FnNameStr, VF, Choices] = *VFInfo;
+
+  llvm::StringRef FnName = FnNameStr;
+  if (!FnName.consume_front("masked_atomicrmw_")) {
+    return std::nullopt;
+  }
+  VectorizationContext::MaskedAtomicRMW AtomicInfo;
+
+  AtomicInfo.VF = VF;
+  AtomicInfo.IsVectorPredicated = Choices.vectorPredication();
+
+  AtomicInfo.IsVolatile = FnName.consume_front("volatile_");
+
+  if (FnName.consume_front("xchg")) {
+    AtomicInfo.BinOp = AtomicRMWInst::BinOp::Xchg;
+  } else if (FnName.consume_front("add")) {
+    AtomicInfo.BinOp = AtomicRMWInst::BinOp::Add;
+  } else if (FnName.consume_front("sub")) {
+    AtomicInfo.BinOp = AtomicRMWInst::BinOp::Sub;
+  } else if (FnName.consume_front("and")) {
+    AtomicInfo.BinOp = AtomicRMWInst::BinOp::And;
+  } else if (FnName.consume_front("nand")) {
+    AtomicInfo.BinOp = AtomicRMWInst::BinOp::Nand;
+  } else if (FnName.consume_front("or")) {
+    AtomicInfo.BinOp = AtomicRMWInst::BinOp::Or;
+  } else if (FnName.consume_front("xor")) {
+    AtomicInfo.BinOp = AtomicRMWInst::BinOp::Xor;
+  } else if (FnName.consume_front("max")) {
+    AtomicInfo.BinOp = AtomicRMWInst::BinOp::Max;
+  } else if (FnName.consume_front("min")) {
+    AtomicInfo.BinOp = AtomicRMWInst::BinOp::Min;
+  } else if (FnName.consume_front("umax")) {
+    AtomicInfo.BinOp = AtomicRMWInst::BinOp::UMax;
+  } else if (FnName.consume_front("umin")) {
+    AtomicInfo.BinOp = AtomicRMWInst::BinOp::UMin;
+  } else if (FnName.consume_front("fadd")) {
+    AtomicInfo.BinOp = AtomicRMWInst::BinOp::FAdd;
+  } else if (FnName.consume_front("fsub")) {
+    AtomicInfo.BinOp = AtomicRMWInst::BinOp::FSub;
+  } else if (FnName.consume_front("fmax")) {
+    AtomicInfo.BinOp = AtomicRMWInst::BinOp::FMax;
+  } else if (FnName.consume_front("fmin")) {
+    AtomicInfo.BinOp = AtomicRMWInst::BinOp::FMin;
+  } else if (FnName.consume_front("uincwrap")) {
+    AtomicInfo.BinOp = AtomicRMWInst::BinOp::UIncWrap;
+  } else if (FnName.consume_front("udecwrap")) {
+    AtomicInfo.BinOp = AtomicRMWInst::BinOp::UDecWrap;
+  } else {
+    return std::nullopt;
+  }
+
+  if (!FnName.consume_front("_align")) {
+    return std::nullopt;
+  }
+
+  uint64_t Alignment = 0;
+  if (FnName.consumeInteger(10, Alignment)) {
```

Comment:
maybe a comment to say what 10 means here?

Reply:
Yep, good shout. It's just the radix.